ligo/src/defaults.mligo

```diff
@@ -41,5 +41,23 @@ let default_startup_storage : startup_storage = {
     stored_entrypoints = (Big_map.empty : stored_entrypoints);
     }
 
-let default_full_storage (admin, token_address : (address * address)) : full_storage =
-  (default_startup_storage, (default_storage (admin, token_address), default_config))
+let create_metadata_map (host_address, host_chain, key : string * string * string) : metadata_map =
+  let address_null = host_address = "" in
+  let chain_null = host_address = "" in
+  let key_null = key = "" in
+  if address_null && chain_null && key_null then
+    (Big_map.empty : metadata_map)
+  else
+    let uri =
+      "tezos-storage://"
+      ^ host_address
+      ^ (if chain_null then "" else host_chain ^ ".")
+      ^ "/"
+      ^ key in
+    Big_map.literal ["", Bytes.pack uri]
```

Comment:
@heitor-lassarote I realized now that this actually doesn't work.

The conversion from string to bytes isn't the same one as `Bytes.pack` per [TZIP-16 specification](https://gitlab.com/tzip/tzip/-/blob/27e556c0ca7f13acbf0eacea354f426447f02f18/proposals/tzip-16/tzip-16.md#contract-storage):
> Unless otherwise-specified, the encoding of the values must be the direct stream
of bytes of the data being stored. For instance, an URI starting with http:
will start with the 5 bytes 0x687474703a (h is 0x68, t is 0x74,
etc.). There is no implicit conversion to Michelson's binary format (PACK) nor
quoting mechanism.

Ligo is unable to make this conversion AFAIK, so this cannot be supported directly in LIGO code (unless it gets added in the future).

So supporting this would only mean adding a dependency just for this (single) TZIP-16 metadata use.
IMO the only thing that we can do here is accept a metadata `big_map` value directly in the `Makefile`.

So change `default_full_storage` to have `metadata_map` as input instead of `metadata_host_address, metadata_host_chain, metadata_key` and in the `Makefile` let's just put:

```
$(OUT)/treasuryDAO_storage.tz : metadata_map = Big_map.empty

	<...>'default_treasury_DAO_full_storage(<...>, "$(metadata_map)")' <...>
```

Reply:
Good observation! Made the changes.